documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,947 @@
+# Migrate from v1 to v2 of the AWS IoT SDK for Java
+
+The AWS IoT Device SDK for Java v2 is a major rewrite of the v1 SDK code base built on top of Java 8+. It includes many updates,
```

Comment:
Trivial: "v1 SDK" -> "AWS IoT Device SDK Java" with embedded link to https://github.com/aws/aws-iot-device-sdk-java

Reply:
"The AWS IoT Device SDK for Java v2 is a major rewrite of the AWS IoT Device SDK for Java code base built on top of Java 8+."
Something like this? Maybe add v1 somewhere?

Reply:
I replaced `v1 SDK` with `AWS IoT Device SDK for Java v1`, so the two versions are more distinguishable in the sentence.